demodoctrine/src/Controller/Admin/QuotesController.php

```diff
@@ -0,0 +1,252 @@
+
+        return $responseBuilder->buildSearchResponse(
+            $this->get('prestashop.module.demodoctrine.grid.definition.factory.quotes'),
+            $request,
+            QuoteGridDefinitionFactory::GRID_ID,
+            'ps_demodoctrine_quote_index'
+        );
+    }
+
+    /**
+     * List quotes
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function generateAction(Request $request)
+    {
+        if ($request->isMethod('POST')) {
+            $generator = $this->get('prestashop.module.demodoctrine.quotes.generator');
+            $generator->generateQuotes();
+            $this->addFlash('success', $this->trans('Quotes were successfully generated.', 'Modules.Demodoctrine.Admin'));
+
+            return $this->redirectToRoute('ps_demodoctrine_quote_index');
+        }
+
+        return $this->render(
+            '@Modules/Demodoctrine/views/templates/admin/generate.html.twig',
+            [
+                'enableSidebar' => true,
+                'layoutTitle' => $this->trans('Quotes', 'Modules.Demodoctrine.Admin'),
+                'layoutHeaderToolbarBtn' => $this->getToolbarButtons(),
+            ]
+        );
+    }
+
+    /**
+     * Create quote
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function createAction(Request $request)
+    {
+        $quoteFormBuilder = $this->get('prestashop.module.demodoctrine.form.identifiable_object.builder.quote_form_builder');
+        $quoteForm = $quoteFormBuilder->getForm();
+        $quoteForm->handleRequest($request);
+
+        $quoteFormHandler = $this->get('prestashop.module.demodoctrine.form.identifiable_object.handler.quote_form_handler');
+        $result = $quoteFormHandler->handle($quoteForm);
+
+        if (null !== $result->getIdentifiableObjectId()) {
+            $this->addFlash(
+                'success',
+                $this->trans('Successful creation.', 'Admin.Notifications.Success')
+            );
+
+            return $this->redirectToRoute('ps_demodoctrine_quote_index');
+        }
+
+        return $this->render('@Modules/Demodoctrine/views/templates/admin/create.html.twig', [
+            'quoteForm' => $quoteForm->createView(),
+        ]);
+    }
+
+    /**
+     * Edit quote
+     *
+     * @param Request $request
+     * @param int $quoteId
+     *
+     * @return Response
+     */
+    public function editAction(Request $request, $quoteId)
+    {
+        $quoteFormBuilder = $this->get('prestashop.module.demodoctrine.form.identifiable_object.builder.quote_form_builder');
+        $quoteForm = $quoteFormBuilder->getFormFor((int) $quoteId);
+        $quoteForm->handleRequest($request);
+
+        $quoteFormHandler = $this->get('prestashop.module.demodoctrine.form.identifiable_object.handler.quote_form_handler');
+        $result = $quoteFormHandler->handleFor((int) $quoteId, $quoteForm);
+
+        if ($result->isSubmitted() && $result->isValid()) {
```

Comment:
what if it fails? Maybe return an error message and stay in edit form.

Reply:
If the form is not valid, the errors will be displayed in the form since we render it, no need for flash error message then